docker/sui-node/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM rust:1.60.0 AS chef
+WORKDIR sui
+RUN apt-get update && apt-get install -y cmake clang
+
+# Plan out the 3rd-party dependencies that need to be built.
+#
+# This is done by:
+#   1. Copy in Cargo.toml, Cargo.lock, and the workspace-hack crate
+#   2. Removing all workspace crates, other than the workpsace-hack
+#      crate, from the workspace Cargo.toml file.
+#   3. Update the lockfile in order to reflect the changes to the
```

Comment:
Doesn't updating the lock file invalidate its purpose, which is to protect versions when being built from inadvertently changing?

Reply:
All of the pinned dependencies are left in tack in the lockfile. this update is done to remove the other workspace members from the lockfile before passing it off to the builder step